src/main.tf

```diff
@@ -167,11 +170,13 @@ resource "google_container_node_pool" "nodes" {
 
   # UPGRADES, REPAIR, AND MAINTENANCE
   management {
-    auto_repair  = true
-    auto_upgrade = true
+    auto_repair = true
+    # this fights the node version if set to true
+    # https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_node_pool#version
+    auto_upgrade = false
```

Comment:
this is the core issue. nodes were being upgraded to 1.22.X and then the next run would downgrade them. Turning this off and setting via the data resource will only trigger node replacements if there are updates... on Terraform runs. It won't happen automatically in the background anymore.